crates/consensus/auto-seal/src/task.rs

```diff
@@ -223,11 +229,25 @@ where
                                 }
                             }
 
-                            let header = header.seal_slow();
-                            debug!(target: "consensus::auto", header=?header.hash(), "sending block notification");
+                            // seal the block
+                            let block = Block {
+                                header,
+                                body: transactions,
+                                ommers: vec![],
+                                withdrawals: None,
+                            };
+                            let sealed_block = block.seal_slow();
+                            let sealed_block_with_senders =
+                                SealedBlockWithSenders::new(sealed_block, senders)
+                                    .expect("senders are valid");
+                            debug!(target: "consensus::auto", header=?sealed_block_with_senders.hash(), "sending block notification");
+
+                            let chain =
+                                Arc::new(Chain::new(vec![(sealed_block_with_senders, post_state)]));
 
                             // send block notification
-                            let _ = new_block_notification_sender.send(Arc::new(header));
+                            let _ = canon_state_notification
+                                .send(reth_provider::CanonStateNotification::Commit { new: chain });
```

Comment:
@mattsse this looks little bit strange for me, maybe it is good for auto-seal but pending blocks should not throw the `new block` notification (or in this case `canon state notif`, but as you already working on miner i didn't want to disrupt anything and just integrated new event.